tchannel/peer_heap.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import absolute_import
+
+from .container import heap
+from .container.heap import HeapOperation
+
+
+class PeerHeap(HeapOperation):
+    def __init__(self):
+        self.peers = []
+
+    def size(self):
+        return len(self.peers)
+
+    def lt(self, i, j):
+        return self.peers[i].score < self.peers[j].score
+
+    def push(self, x):
+        x.index = len(self.peers)
+        self.peers.append(x)
+
+    def pop(self):
+        n = len(self.peers)
+        item = self.peers[n-1]
+        item.index = -1
+        del self.peers[n - 1]
```

Comment:
This can be compressed into:

    item = self.peers.pop()
    item.index = -1
    return item